### DISCUSSION TITLE
Remix will take a nap?

### DISCUSSION THREAD

**Discussion by danielehrhardt** [remix-run/remix#9439] [General]
Hi! :) 

i am currently watching the React Conf and Ryan Florence said that Remix will be merged to React Router and Remix itself will take a nap. 

So i am confused when Developing Shopify Apps what to use. 

Can we maybe discuss this?

Best
Daniel

**Comment by alexanderson1993** [comment]
There's a section at the bottom of this blog post that explains: https://remix.run/blog/merging-remix-and-react-router

- For now, keep using Remix
- Get your Remix app ready to upgrade to Remix v3 / React Router v7 with future flags
- Replace your Remix dependencies and imports to point to React Router
- Tada! You're all set.

**Comment by kiliman** [comment]
It's just a name change. Keep doing what you're doing now. When RR 7 is released, there will be an easy migration process.

**Reply by danielehrhardt** [reply]
This is not the Problem i have. It's more like if you should start with a dying project. 
When developing Shopify Plugins then you will setup it up with some base structure, remix.run and this is somehow wrong if the Project will take a nap. 

https://shopify.dev/docs/apps/getting-started/create

**Reply by alexanderson1993** [reply]
That's all FUD - Fear, Uncertainty, Doubt. 

React Router v7 will work just as well as Remix v3. It's as if they were exactly the same package.

In fact, by all measurements, they will be exactly the same except in name. So, the team has three options:
- Keep React Router and Remix as two separate projects, basically doubling the maintenance effort and artificially splitting users.
- Stick with the Remix name, have all the React Router users migrate to using the package called Remix, and retire the React Router name.
- Have all the Remix users migrate to using the package named React Router, and reserve the Remix name for some other exciting features in the future.

They went with option 3. I know the Remix team has been weighing this decision for months now. They don't make it lightly. I don't think we should take it lightly - discussion is welcome. But the more I've dug into it, the more this seems like the best decision for Remix and React Router as a whole.

Going back to the first point, it's worth remembering: this is just a name change. The underlying tool and capabilities are all the same that we've grown to know and love.

**Reply by danielehrhardt** [reply]
![image](https://github.com/remix-run/remix/assets/19570043/5e1144c5-91d9-4f32-9cd5-8b3be40fe6a1)

**Comment by bluefire2121** [comment]
> That's all FUD - Fear, Uncertainty, Doubt.
> 
> React Router v7 will work just as well as Remix v3. It's as if they were exactly the same package.
> 
> In fact, by all measurements, they will be exactly the same except in name. So, the team has three options:
> 
> * Keep React Router and Remix as two separate projects, basically doubling the maintenance effort and artificially splitting users.
> * Stick with the Remix name, have all the React Router users migrate to using the package called Remix, and retire the React Router name.
> * Have all the Remix users migrate to using the package named React Router, and reserve the Remix name for some other exciting features in the future.
> 
> They went with option 3. I know the Remix team has been weighing this decision for months now. They don't make it lightly. I don't think we should take it lightly - discussion is welcome. But the more I've dug into it, the more this seems like the best decision for Remix and React Router as a whole.
> 
> Going back to the first point, it's worth remembering: this is just a name change. The underlying tool and capabilities are all the same that we've grown to know and love.

Try convincing eleventy-hundred something stakeholders that it's just a name change in a large corporate environment. I can say it's just a name change all day. It'll still require new marketing material, new communications, new assessments, new reviews, new messaging for which leadership will need to align, new education, new security checks, etc. I get that between Remix and React Router, one side has to "feel the pain and do the work", but nevertheless; "it's still pain and it's still work".

**Reply by sergiodxa** [reply]
Out of curiosity, why do you need new marketing materials?

**Reply by bluefire2121** [reply]
Globally we have material that localizes the word Remix in several different languages in a number of getting started documents. Even a simple change from Remix to React-Router requires new internal marketing material to explain the change, convince people the change is safe, convince people the change is warranted, convince people the change is easy so forth and so forth. We have a large global, developer population that requires step-by-step instruction.

I get it. Remix is outnumbered so it's easier to tell a few Remix users to move back to React Router than it is to tell all the hesitant React Router users to move to Remix.
